### PR TITLE
build: cpack: Create macOS installer

### PR DESCRIPTION
**Pull request by cosmo0920**
Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

Currently, we don't provide macOS installer such as pkg style.
This PR makes to be able to create macOS installer.

Nowadays, macOS platform has two architectures, x86_64(intel) and arm64 (apple silicon).
In this PR, we also have to handle cpu input plugin absence on macOS. (It is for Linux platform only.)

For Apple Silicon (macOS-arm64), cpack will be append `-apple` suffix on fluent-bit package name.
For Intel Mac (macOS-x86_64), cpack will be append `-intel` suffix on fluent-bit package name.

None of relevant issues.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:
- [N/A] Example configuration file for the change
- [ ] Debug log output from testing the change
This PR can be testing as:
```console
$ cd build
$ cmake -DCPACK_GENERATOR=productbuild .. 
$ make -j $(sysctl -n hw.ncpu)
$ cpack -G productbuild
```

- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.
- [ ] Attached [local packaging test](./packaging/local-build-all.sh) output showing all targets (including any new ones) build.

**Documentation**

- [x] Documentation required for this feature

Might be required and rewrite installation description on macOS part.

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -988,6 +988,7 @@ endif()
 # Enable components
 set(CPACK_DEB_COMPONENT_INSTALL ON)
 set(CPACK_RPM_COMPONENT_INSTALL ON)
+set(CPACK_productbuild_COMPONENT_INSTALL ON)
```
Comment (patrick-stephens) [inline CMakeLists.txt:991]: Should this be capitalised?

Comment (cosmo0920) [reply CMakeLists.txt:991]: I think this should be corresponding to generator name.

On macOS, supported generators are:

```console
% cpack -G unknown
CPack Error: Could not create CPack generator: unknown

Generators
  7Z                           = 7-Zip file format
  Bundle                       = Mac OSX bundle
  DragNDrop                    = Mac OSX Drag And Drop
  External                     = CPack External packages
  IFW                          = Qt Installer Framework
  NSIS                         = Null Soft Installer
  NSIS64                       = Null Soft Installer (64-bit)
  NuGet                        = NuGet packages
  PackageMaker                 = Mac OSX Package Maker installer
  STGZ                         = Self extracting Tar GZip compression
  TBZ2                         = Tar BZip2 compression
  TGZ                          = Tar GZip compression
  TXZ                          = Tar XZ compression
  TZ                           = Tar Compress compression
  TZST                         = Tar Zstandard compression
  ZIP                          = ZIP file format
  productbuild                 = Mac OSX pkg
```

**Comment by patrick-stephens** [comment]
Do we need to update the docs as well @cosmo0920 ?

**Comment by cosmo0920** [comment]
Yeah, we need to update the docs as well after starting to provide macOS installer.

**Comment by patrick-stephens** [comment]
Intel only as it is what Github supports, anything more is complex and requires separate effort on CI to handle so a separate PR later.

We can indicate how to build of course and that we provide pre built binary for a specific architecture.

**Comment by cosmo0920** [comment]
I added the instruction section for building macOS installer from source: https://github.com/fluent/fluent-bit-docs/pull/781

**Comment by patrick-stephens** [comment]
Sounds good to me, LGTM

**Review by niedbalski — APPROVED** [review]
LGTM

**Comment by edsiper** [comment]
@cosmo0920 can you submit pls another pr to move the macos setup to cmake/macos-setup.cmake ?

**Pull request merged**